typegen/generate-modules.js

```diff
@@ -0,0 +1,34 @@
+const fs = require('fs');
+const path = require('path');
+
+function generateSubModules(mainJsFile, basePath) {
+  const pathMap = new Map();
+  const regex = /export\s+(\S+|\{[\s\w,]*\})\s+from\s+'(\.\/)?(\S+)'/g;
+
+  let match;
+  do {
+    match = regex.exec(mainJsFile);
+    if (match) {
+      pathMap.set(match[1], match[3]);
+    }
+  } while (match);
+
+  Array.from(pathMap).forEach(([name, componentPath]) => {
+    let fullPath = path.join(basePath, `${componentPath}`);
+    fullPath = fs.existsSync(fullPath) && fs.statSync(fullPath).isDirectory()
```

Comment:
can you make this conditional a named boolean so that future reads can understand what it is doing a little easier?